x-pack/docs/en/security/authentication/configuring-kerberos-realm.asciidoc

```diff
@@ -0,0 +1,182 @@
+default realm, the Key Distribution Center (KDC), and other configuration details 
+required for Kerberos authentication. When the JVM needs some configuration 
+properties, it tries to find those values by locating and loading this file. The 
+JVM system property to configure the file path is `java.security.krb5.conf`. To 
+configure JVM system properties see {ref}/jvm-options.html[configuring jvm options]. 
+If this system property is not specified, Java tries to locate the file based on 
+the conventions.
+
+TIP: It is recommended that this system property be configured for {es}.
+The method for setting this property depends on your Kerberos infrastructure. 
+Refer to your Kerberos documentation for more details.
+
+For more information, see http://web.mit.edu/kerberos/krb5-latest/doc/admin/conf_files/krb5_conf.html[krb5.conf]
+
+--
+
+. Create a keytab for the service node. 
++
+--
+A keytab is a file that stores pairs of principals and encryption keys. {es} 
+uses the key from the keytab to decrypt the tickets presented by the user. You 
+must create a keytab for {es} by using the tools provided by your Kerberos 
+implementation. For example, some tools that create keytabs are `ktpass.exe` on 
+Windows and `kadmin` for MIT Kerberos. 
+--
+
+. Put the keytab file in the `ES_PATH_CONF`.
++
+--
+Make sure that this keytab file has read permissions. This file contains
+credentials, therefore you must take appropriate measures to protect it.
+
+IMPORTANT: On every {es} node, there must be a keytab file for the HTTP
+principal. The keytab files are unique for each node since they include the
+hostname. Usually it will be of format `HTTP/es.domain.local@ES.DOMAIN.LOCAL`.
+An {es} node can act as any principal a client requests as long as that
+principal and its credentials are found in the configured keytab.
+
+--
+
+. Create a Kerberos realm. 
++
+--
+
+To enable Kerberos authentication in {es}, you must add a Kerberos realm in the 
+realm chain.
+
+NOTE: You can configure only one Kerberos realm on {es} nodes.
+
+To configure a Kerberos realm, there are few mandatory realm settings and
+other optional settings that you need to configure in the `elasticsearch.yml`
+configuration file. Add a realm of type `kerberos` under the 
+`xpack.security.authc.realms` namespace.
+
+The most common configuration for a Kerberos realm is as follows:
+
+[source, yaml]
+------------------------------------------------------------
+xpack.security.authc.realms.kerb1:
+  type: kerberos
+  order: 3
+  keytab.path: es.keytab
+  remove_realm_name: false
+------------------------------------------------------------
+
+The `username` is extracted from the ticket presented by user and usually has 
+the format `username@REALM`. This `username` can then be used for mapping
+roles to the user. There is a realm setting `remove_realm_name` which when
+set to `true` removes the realm part (`@REALM`) and the resulting `username` can 
+then be used for role mapping.
```

Comment:
Per previous comment, this seems to imply that you can't use role mappings successfully unless you set remove_realm_name to true.  Is that right?